library/posture.c

```diff
@@ -545,3 +585,273 @@ static void ziti_pr_handle_process(ziti_context ztx, char *id, char *path, bool
 
     ziti_collect_pr(ztx, path, obj, obj_len);
 }
+
+static void default_pq_os(ziti_context ztx, const char *id, ziti_pr_os_cb response_cb) {
+    const char *os;
+    const char *ver;
+    const char *build;
+#if _WIN32
+    OSVERSIONINFOEXW os_info = {0};
+    os_info.dwOSVersionInfoSize = sizeof(os_info);
+    if (pRtlGetVersion) {
+        pRtlGetVersion((PRTL_OSVERSIONINFOW) &os_info);
+    } else {
+        /* Silence GetVersionEx() deprecation warning. */
+#pragma warning(suppress : 4996)
+        GetVersionExW(&os_info);
+    }
+
+    switch (os_info.wProductType) {
+        case 1: os = "windows";
+            break;
+        case 2:
+        case 3: os = "windowsserver";
+            break;
+        default:
+            os = "<unknown windows type>";
+    }
+    char winver[16];
```

Comment:
while this definitely seems like enough why not just make it bigger just because... it's short lived anyway. i worry :)

Reply:
32?

Reply:
it won't crash, even if  the buffer is too short

Reply:
i just worry that it'll be a hard to track down bug is all. each of those fields "can be" up to like 10 characters long as a string, right? if MS makes them crazy long some day on purpose or by mistake it'd be a problem. 32 seems like it'd be enough but why not just go ham and use 64?